### DISCUSSION TITLE
Dynamic fields based on selections from 2 multi select dropdowns

### DISCUSSION THREAD

**Discussion by afraazali** [filamentphp/filament#9203] [Help]
I've got a simple form with the following fields:

* A multi select field with options
  * A
  * B
  * C
* A multi select field with the options
  * 1
  * 2
  * 3
 
When a user chooses a combination of of these fields, I'd like to generate a table with inputs for each combination. For example, if the user chooses A and B for dropdown 1 and selects 2 and 3 for dropdown 2, the given table would be created:

|   | A    | B |
| -------- | -------- | ------- |
| 2  | texfield   | texfield   |
| 3  | texfield      | texfield    |

Kind of like an excel sheet. I'd like to then take the values from the textfields and store it in a JSON field on the model. The part that I'm unclear on, is allowing the user to sort the selected items. So if they wanted have B displayed before A in the grid/table, they'd be able to sort the table rows/columns without losing any of the data in the respective texfields. Also being able to select additional options from the dropdowns should not remove any of the data that's been input in the textfields in the table.

I thought to do this with a View component, but realized that with a reactive approach, the table would be re-rendered on each select choice, basically blanking out the textfields. I'm not quite sure if a custom field is even the right approach here. Any help is greatly appreciated!

**Comment by leandrocfe** [comment]
what code did you try?

**Reply by afraazali** [reply]
```php
View::make('forms.components.table')
    ->dehydrated(false)
    ->schema(function (Get $get): array {
        $rows = $get('data.one') ?? [];
        $columns = $get('data.two') ?? [];
        $schema = [];

        foreach ($rows as $row) {
            foreach ($columns as $column) {
                $schema[] = TextInput::make("data.{$row}.{$column}")
                    ->meta('row', $row)
                    ->meta('column', $column)
                    ->extraInputAttributes(['size' => 5])
                    ->label('');
            }
        }

        return $schema;
    })
```

In my blade component (it's a mess):

```php
@php
    $childComponents = $getChildComponentContainer()->getComponents();

    $rows = array_unique(array_map(fn(\Filament\Forms\Components\Component $component) => $component->getMeta('row'), $childComponents));
    $columns = array_unique(array_map(fn(\Filament\Forms\Components\Component $component) => $component->getMeta('column'), $childComponents));

    $componentMap = array_reduce($childComponents, function ($prev, $curr) {
      $prev["{$curr->getMeta('row')}_{$curr->getMeta('column')}"] = $curr;
      return $prev;
    }, []);
@endphp

<div>
    <table class="table-auto border mt-8">
        <tbody>
        @foreach($row as $row)
            <tr>
                <td class="border p-2">{{ $row }}</td>
                @foreach($column as $column)
                    <td class="border p-2">
                        <div class="flex justify-center">
                            {{ $componentMap["{$column}_$row"] }}
                        </div>
                    </td>
                @endforeach
            </tr>
        @endforeach
        </tbody>
    </table>
</div>
```

Allowing the user to sort the rows/columns of the table hasn't been started yet, but I'm sure there's a better way to do this. The final json output that I'm trying to store in the database would be:

```
{
  row1: {column1: "input value", column2: "input value"},
  row2: {column1: "input value", column2:  "input value"}
}
```

So taking the example from above (A, B, 2, 3):

```
{
  A: {2: "input value", 3: "input value"},
  B: {2: "input value", 3: "input value"}
}
```

Of course if they sort the table, that sorting needs to be respected in the JSON, which might be another ordeal.
```
